src/components/tutorials-sidebar/index.tsx

```diff
@@ -49,32 +52,78 @@ function CollectionViewSidebarContent({
 					<Fragment key={title}>
 						<HorizontalRule />
 						{title ? <SectionTitle text={title} /> : null}
-						<SectionList items={items} />
+						<SectionList>
+							{items.map(({ text, href, isActive }: ListItemProps) => {
+								return (
+									<ListItem
+										key={`${text}${href}`}
+										text={text}
+										href={href}
+										isActive={isActive}
+									/>
+								)
+							})}
+						</SectionList>
 					</Fragment>
 				)
 			})}
 		</>
 	)
 }
 
-function TutorialViewSidebarContent({ items }: SectionListProps) {
-	return <SectionList items={items} />
+function TutorialViewSidebarContent({
+	items,
+}: {
+	items: TutorialListItemProps[]
+}) {
+	return (
+		<SectionList>
+			{items.map(
+				({
+					text,
+					href,
+					isActive,
+					tutorialId,
+					collectionId,
+				}: TutorialListItemProps) => {
+					return (
+						<TutorialListItem
+							key={`${collectionId}${tutorialId}`}
+							text={text}
+							href={href}
+							isActive={isActive}
+							tutorialId={tutorialId}
+							collectionId={collectionId}
+						/>
+					)
+				}
+			)}
+		</SectionList>
+	)
+}
+
+function SectionList({ children }: SectionListProps) {
+	return <ul className={s.listRoot}>{children}</ul>
 }
 
-function SectionList({ items }: SectionListProps) {
+function TutorialListItem({
+	href,
+	isActive,
+	text,
+	tutorialId,
+	collectionId,
+}: TutorialListItemProps) {
+	/**
+	 * Query for progress, and display the appropriate status icon
+	 */
+	const { tutorialProgressStatus } = useTutorialProgress({
```

Comment:
🤔 Instead of each tutorial calling for its own progress data here. I think it would make more sense to call for the entire [collection's progress state](https://digital-api-specs.vercel.app/learn#tag/Collection-Tutorial-Progress/paths/~1collections~1{id}~1progress/get). That would make it one single call. And then the progress status would be passed down to these children. Thoughts?

Reply:
[See related comment here ](https://github.com/hashicorp/dev-portal/pull/900/files#r961006560)

Reply:
@kendallstrautman @ashleemboyer ☝️ yep, exactly - it seems like with `react-query`, compared to direct API requests, rather than have to choose "should we do this individually or in batch" I think we can do both. Writing this out cause I'm still building my understanding of it:
- individual components that need to data always make their individual query. Ensures they have the data they need, and avoids prop drilling... but means a ton of separate requests as a result of those queries if we do nothing else.
- At the level of any view or component that renders many individual query-data-needing components, we can optimize the # of requests by making a batch query. My understanding is this populates the `react-query` cache so that all the lil subsequent individual queries don't result in a ton of separate requests.

Reply:
@zchsh yep, you're right on target! 🎯